ddtrace/llmobs/_integrations/crewai.py

```diff
@@ -247,6 +274,117 @@ def _get_agent_tools(self, tools):
+            span_dict = self._flow_span_to_method_to_span_dict.get(span._parent, {}).setdefault(str(response), {})
+            span_dict.update({"span_id": str(span.span_id), "trace_id": format_trace_id(span.trace_id)})
+
+        span._set_ctx_items(
+            {
+                NAME: span.name or "Flow Method",
+                INPUT_VALUE: input_dict,
+                OUTPUT_VALUE: str(response),
+                SPAN_LINKS: span_links,
+            }
+        )
+        return
+
+    def _llmobs_set_span_link_on_flow(self, flow_span, args, kwargs, flow_instance):
+        """
+        Set span links for the next queued listener method(s) in a CrewAI flow.
+
+        Notes:
+            - Router methods passed by name are skipped (span links are based on router results)
+            - AND conditions:
+                - temporary output->output span links added by default for all trigger methods
+                - once all trigger methods have run for the listener, remove temporary output->output links and
+                  add span links from trigger spans to listener span
+        """
+        trigger_method = get_argument_value(args, kwargs, 0, "trigger_method", optional=True)
+        if not self.llmobs_enabled or not trigger_method:
+            return
+        flow_methods_to_spans = self._flow_span_to_method_to_span_dict.get(flow_span, {})
+        trigger_span_dict = flow_methods_to_spans.get(trigger_method)
+        if not trigger_span_dict or trigger_span_dict.get("is_router_method_name", False):
+            return
+        listeners = getattr(flow_instance, "_listeners", {})
+        triggered = False
+        for listener_name, (condition_type, listener_triggers) in listeners.items():
+            if trigger_method not in listener_triggers:
+                continue
+            span_dict = flow_methods_to_spans.setdefault(listener_name, {})
+            span_dict["trace_id"] = format_trace_id(flow_span.trace_id)
+            span_links = span_dict.setdefault("span_links", [])
+            if condition_type != "AND":
+                triggered = True
+                span_links.append(
+                    {
+                        "span_id": str(trigger_span_dict["span_id"]),
+                        "trace_id": format_trace_id(flow_span.trace_id),
+                        "attributes": {"from": "output", "to": "input"},
+                    }
+                )
+                continue
+            if any(
+                flow_methods_to_spans.get(_trigger_method, {}).get("span_id") is None
+                for _trigger_method in listener_triggers
+            ):  # skip if not all trigger methods have run (span ID must exist) for AND listener
+                continue
+            triggered = True
+            for method in listener_triggers:
+                method_span_dict = flow_methods_to_spans.get(method, {})
+                if not method_span_dict:
+                    continue
```

Comment:
could be reading it wrong, but given the `any` check above i don't think we'd ever hit this case